### ISSUE TITLE
generate a certificate with SANs in the documentation

### ISSUE THREAD

**Issue by glyph** [pyca/cryptography#3314]
https://cryptography.io/en/latest/x509/reference/#x-509-certificate-builder generates a certificate that only has a CN.  Given that CN host validation is deprecated and Python warns noisily about it, it would be more useful to document the generation of a certificate with SANs.

**Comment by reaperhulk** [comment]
@glyph do you see that example as a likely way a user would generate a certificate to use in a development environment? I just want to be sure I understand the audience we want to alter the example for. The whole x509 tutorial section seems like it could use substantial expansion, which we could then link to from the reference for more extensive explanation. But I also have basically no ambition to write it...

**Comment by glyph** [comment]
@reaperhulk -

> do you see that example as a likely way a user would generate a certificate to use in a development environment?

Indeed I do!  And in fact I did.  See here, for example - https://github.com/twisted/twisted/commit/98a3df200968b78bd3b985dfd4fb10a5b415d6fc#diff-a0749b8aa4682956a79714cc4a69efd8R200 - where I recently used this example to do exactly that.  Luckily I was using it specifically to address this problem, and i knew I had to go looking for `x509.SubjectAlternativeName` myself.

> The whole x509 tutorial section seems like it could use substantial expansion, which we could then link to from the reference for more extensive explanation

I agree, but I also don't have the motivation to do that.  Nevertheless, I think that including `SubjectAlternativeName` in the simple example is almost as important as including `BasicConstraints`; this change is important even without a more substantive overhaul.
